### PR TITLE
Fix HelpInfoUri and add regression tests for help

### LINKED ISSUE #269: Update-Help doesn't work due to HTTPS in HelpInfoUri
Update-Help doesn't work due to HTTPS in HelpInfoUri. This was fixed in #82 and broken again in #195.

```
Update-Help : Failed to update Help for the module(s) :
'PowerShellGet'
The HelpInfoURI https://go.microsoft.com/fwlink/?linkid=855963 does not start with HTTP.
At line:1 char:1
+ Update-Help -Module PowerShellGet
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (HelpInfoUri:Uri) [Update-Help], Exception
    + FullyQualifiedErrorId : InvalidHelpInfoUriFormat,Microsoft.PowerShell.Commands.UpdateHelpCommand
```

### PR DESCRIPTION
**Pull request by bmanikm**
Resolves #269

### CODE REVIEW

File: Tests/Pester.PSGetModuleHelp.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Path
+    )
+
+    Get-ChildItem -Path $Path -Include $FileType -Recurse -ErrorAction SilentlyContinue | Select-Object -ExpandProperty FullName
+}
+
+Describe 'Validate PowerShellGet module help' -tags 'BVT', 'InnerLoop' {
```
Comment (edyoung) [inline Tests/Pester.PSGetModuleHelp.Tests.ps1]: how long does this take? Seems like it reasonably go in the P2 section if it's more than a couple seconds. #Resolved

Comment (bmanikm) [reply Tests/Pester.PSGetModuleHelp.Tests.ps1]: Sure, I will change the priority to P2 once CI test passes with these regression tests.

---
In reply to: [187751860](https://github.com/PowerShell/PowerShellGet/pull/274#discussion_r187751860) [](ancestors = 187751860)

Comment (bmanikm) [reply Tests/Pester.PSGetModuleHelp.Tests.ps1]: These tests are taking less than a couple seconds.

---
In reply to: [187752827](https://github.com/PowerShell/PowerShellGet/pull/274#discussion_r187752827) [](ancestors = 187752827,187751860)

Comment (bmanikm) [reply Tests/Pester.PSGetModuleHelp.Tests.ps1]: New help tests passed in AppVeyor and TravisCI test runs. I have updated the priority to P1. #Resolved

File: Tests/Pester.PSGetModuleHelp.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+$script:HelpInstallationPath = Join-Path -Path $script:PowerShellGetModuleInfo.ModuleBase -ChildPath 'en-US'
+$script:SaveHelpPath = Join-Path -Path ([System.IO.Path]::GetTempPath()) -ChildPath "PSGetHelp_$(Get-Random)"
+
+function GetFiles {
+    param (
+        [Parameter()]
+        [string]
+        $FileType = "*help.xml",
```
Comment (farehar) [inline Tests/Pester.PSGetModuleHelp.Tests.ps1]: *help.xml isn't really a file type. Can probably rename to FileName. #Resolved

File: Tests/Pester.PSGetModuleHelp.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    It 'Validate Update-Help for the PowerShellGet module' {
+        $UpdateHelp_Params = @{
+            Force = $true
+            UICulture = 'en-US'
+        }
+        if($PSVersionTable.PSVersion -gt '4.0.0') {
+            $UpdateHelp_Params['FullyQualifiedModule'] = $script:FullyQualifiedModuleName
+
+            if($PSVersionTable.PSVersion -gt '6.0.99') {
```
Comment (farehar) [inline Tests/Pester.PSGetModuleHelp.Tests.ps1:56]: Why do we need to add the scope? #Resolved

File: tools/build.psm1 [hunk 4]
```diff
@@ -100,8 +100,15 @@ function Install-PackageManagement {
...
             Install-PackageProvider -Name NuGet -Force
         }
-
-        $LatestOneGetInPSGallery = Find-Module -Name $OneGetModuleName
+        
+        $FindModule_params = @{
+            Name = $OneGetModuleName
+        }
+        if ($PSVersionTable.PSVersion -eq '5.1.14394.1000') {
+            # Adding -MaximumVersion 1.1.7.0 as AppVeyor VM with WMF 5 is not installed with latest root CA certificates
```
Comment (farehar) [inline tools/build.psm1:108]: Wouldn't adding the latest certs to the VM a better fix? Is it hard to get latest root Certs on VMs? #Resolved

File: Tests/Pester.PSGetModuleHelp.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,89 @@
+# This is a Pester test suite to validate the PowerShellGet module help
+#
+# Copyright (c) Microsoft Corporation
+
+$script:IsWindowsOS = (-not (Get-Variable -Name IsWindows -ErrorAction Ignore)) -or $IsWindows
```
Comment (farehar) [inline Tests/Pester.PSGetModuleHelp.Tests.ps1:5]: Is this condition correct. What happens if IsWindows does not exist. We will set the variable to true. I don't think we want that. #Resolved

File: Tests/Pester.PSGetModuleHelp.Tests.ps1 (hunk 3 above)
Comment (bmanikm) [reply Tests/Pester.PSGetModuleHelp.Tests.ps1:56]: -Scope is required in PS 6.1 for installing the help to the module base. This is a recent change in PSCore.

---
In reply to: [188069241](https://github.com/PowerShell/PowerShellGet/pull/274#discussion_r188069241) [](ancestors = 188069241)

File: tools/build.psm1 (hunk 4 above)
Comment (bmanikm) [reply tools/build.psm1:108]: That is not controlled by us. AppVeyor has to update their VM image. This is a temporary workaround for running the CI test runs on WMF 5.
This condition will be removed when we update minimum required version of PackageManagement. Note that other CI test runs are installing the latest version of PackageManagement.

---
In reply to: [188070267](https://github.com/PowerShell/PowerShellGet/pull/274#discussion_r188070267) [](ancestors = 188070267)

File: Tests/Pester.PSGetModuleHelp.Tests.ps1 (hunk 5 above)
Comment (bmanikm) [reply Tests/Pester.PSGetModuleHelp.Tests.ps1:5]: Yes, correct.
 -ErrorAction Ignore takes of the scenario where IsWindows variable is not available.

---
In reply to: [188070651](https://github.com/PowerShell/PowerShellGet/pull/274#discussion_r188070651) [](ancestors = 188070651)

**Review by farehar — APPROVED** [review]

**Pull request merged**